data-service-model/src/main/java/com/amazonaws/blox/dataservicemodel/v1/client/DataService.java

```diff
@@ -35,4 +41,16 @@ CreateEnvironmentResponse createEnvironment(CreateEnvironmentRequest request)
   StartDeploymentResponse startDeployment(StartDeploymentRequest request)
       throws EnvironmentNotFoundException, EnvironmentVersionNotFoundException,
           EnvironmentVersionOutdatedException, InvalidParameterException, ServiceException;
+
+  /** Lists all environments that target a given cluster */
+  ListEnvironmentsResponse listEnvironments(ListEnvironmentsRequest request);
+
+  /** Lists all clusters managed by blox */
+  ListClustersResponse listClusters(ListClustersRequest request);
+
+  /**
+   * Describes the revision of the given EnvironmentRevision that is the current deployment Target.
+   */
+  DescribeEnvironmentRevisionResponse describeTargetEnvironmentRevision(
```

Comment:
should the request and response objects have the same name? so describeTargetEnvironmentResponse instead of describeEnvironmentRevisionResponse?

Reply:
Yeah, I was wondering about that. It looks like both `describeTargetEnvironmentRevision` and `describeEnvironmentRevision` would return exactly the same response, but I never ended up needing the `describeEnvironmentRevision` API call, so I removed it from the interface. I'll rename this, and we can rename/reuse the type later if needed.